### PR TITLE
feat(switch): add component tokens

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #7180 

## Summary

Adds the following component tokens (CSS props):

* `--calcite-switch-handle-background-color`
* `--calcite-switch-handle-border-color`
* `--calcite-switch-track-background-color`
* `--calcite-switch-track-border-color`
* `--calcite-switch-corner-radius`
* `--calcite-switch-shadow`

### CODE REVIEW

File: packages/calcite-components/src/components/switch/switch.scss [hunk 1]
```diff
@@ -1,125 +1,131 @@
...
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-switch-handle-background-color: Specifies the background color of the component handle.
+ * @prop --calcite-switch-handle-border-color: Specifies the border color of the component handle.
+ * @prop --calcite-switch-track-background-color: Specifies the background color of the component track.
+ * @prop --calcite-switch-track-border-color: Specifies the border color of the component track.
+ * @prop --calcite-switch-corner-radius: Specifies the border radius of the component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/switch/switch.scss:10]: Specifies the "corner radius"

File: packages/calcite-components/src/components/switch/switch.scss [hunk 2]
```diff
@@ -1,125 +1,131 @@
...
+ * @prop --calcite-switch-shadow: Specifies the box shadow of the component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/switch/switch.scss:11]: Specifies the "shadow" (vs. box shadow), I think.

**Review by alisonailea — APPROVED** [review]

**Comment by macandcheese** [comment]
There are some "invisible" height changes in Chromatic, lets make sure this won't affect layout somehow: https://www.chromatic.com/test?appId=6266d45509d7eb004aa254fb&id=65d535a602d615a0c2d7deba

**Comment by jcfranco** [comment]
Forgot to bring this up sooner, but name-wise, do we prefer `handle` vs `thumb`? This might be the first instance of handle/thumb in a public property name (related to that UI part). I'm used to "thumb" and is slightly shorter, but wanted to confirm for consistency. cc @geospatialem @DitwanP

**Comment by geospatialem** [comment]
> Forgot to bring this up sooner, but name-wise, do we prefer `handle` vs `thumb`? This might be the first instance of handle/thumb in a public property name (related to that UI part). I'm used to "thumb" and is slightly shorter, but wanted to confirm for consistency.

We use `handle` for the `list-item` event, so lean towards `handle`, unless we want to differentiate between the list event and switch prop.
https://github.com/Esri/calcite-design-system/blob/802ccf2272985d31bd5afa2383c24e669913ee5c/packages/calcite-components/src/components/list-item/list-item.tsx#L249-L251

**Comment by driskull** [comment]
I prefer handle since thumb might be confused with thumbnail.

**Pull request merged**